### DISCUSSION TITLE
Is PHP 8.2 not supported only 8.1?

### DISCUSSION THREAD

**Discussion by bomshteyn** [NativePHP/laravel#200] [Q&A] [answered]
When  running `php artisan native:serve` 
I get the following error:
```
Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.2.0". You are running 8.1.17. in /Users/User/PhpstormProjects/zms/vendor/composer/platform_check.php on line 24
```

Is there a way to have it work with PHP 8.2?

**Comment by pb30** [comment]
NativePHP supports 8.2. But per the error message:

> You are running 8.1.17

Your composer.json has a requirement for 8.2 but you are running 8.1.

**Reply by bomshteyn** [reply]
When i do `php -v`

Here is the version i get:
```
PHP 8.2.9 (cli) (built: Aug 16 2023 21:06:19) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.9, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.9, Copyright (c), by Zend Technologies
```

How do i switch the version nativePHP uses?

**Reply by pb30** [reply]
Are you getting that Composer error in your app or in your terminal?

**Reply by bomshteyn** [reply]
in terminal

**Reply by pb30** [reply]
Any difference if you run `composer install`, then try again?

**Reply by bomshteyn** [reply]
no, same issue

**Reply by pintend** [reply]
NativePHP doesnt use the php version on your system but uses a static build

The static php build included with NativePHP is php 8.1

https://github.com/NativePHP/php-bin/blob/main/bin/mac/arm64/php which is downlaoded to ./vendor/nativephp/electron/resources/js/resources/php/php is php 8.1

**Comment by pintend** [comment]
If you are on a mac and use Herd you can use their static php build

From the root of your project:
`cp $HOME/Library/Application\ Support/Herd/bin/php82 ./vendor/nativephp/electron/resources/js/resources/php/php`

**Reply by MikeCraig418** [reply]
As per https://github.com/NativePHP/laravel/issues/242, I needed to replace the binaries in `./vendor/nativephp/php-bin/bin/mac`

**Comment by simonhamp** [comment, marked as answer]
As of [`nativephp/php-bin@0.4.0`](https://github.com/NativePHP/php-bin/releases/tag/0.4.0), PHP8.2 _and_ 8.3 are now supported across all platforms.

Enjoy!
